src/ray/common/ray_config_def.h

```diff
@@ -506,6 +507,11 @@ RAY_CONFIG(uint64_t, task_events_send_batch_size, 10 * 1000)
 /// Setting the value to -1 allows unlimited profile events to be sent.
 RAY_CONFIG(int64_t, task_events_max_num_profile_events_for_task, 1000)
 
+/// Max number of task attempts being dropped on the worker side to report to GCS.
+/// Setting the value to -1 allows unlimited dropped task attempts in a single
+/// report to GCS.
+RAY_CONFIG(int64_t, task_events_drop_task_attempt_batch_size, 10 * 1000)
```

Comment:
```suggestion
RAY_CONFIG(int64_t, task_events_dropped_task_attempt_batch_size, 10 * 1000)
```

Also can you explain why this needs to be batched? (because we should report task attempt to GCS for correct accounting)

Reply:
Also this means each report size will increased. Would this be small enough? (Can you calculate it?)

Reply:
> Also can you explain why this needs to be batched? 

We will need to let GCS know what have been dropped on the worker (otherwise GCS will not be able to do accounting properly) 

> Also this means each report size will increased. Would this be small enough? (Can you calculate it?)

This is exactly the reason why it's batched actually, so 10K of (task id + attempt number) => 10K of (24B + 8B)  => 10K * 32B ~=> 320KiB